### PR TITLE
rule: add --query.http-method

### PR DESCRIPTION
**Pull request by GiedriusS**
Add a new parameter which lets you specify what kind of HTTP method to
use when sending queries.

Switch to using POST by default to avoid possible problems and because it
has been around for quite a while by now.

Signed-off-by: Giedrius Statkevičius <giedriuswork@gmail.com>

* [x] I added CHANGELOG entry for this change.
* [ ] Change is not relevant to the end user.

## Verification

Tests still pass.

Closes https://github.com/thanos-io/thanos/issues/2577.
Closes https://github.com/thanos-io/thanos/pull/3233.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,7 @@ We use *breaking :warning:* to mark changes that are not backward compatible (re
 - [#3346](https://github.com/thanos-io/thanos/pull/3346) Ruler UI: Fix a bug preventing the /rules endpoint from loading.
 - [#3115](https://github.com/thanos-io/thanos/pull/3115) compact: now deletes partially uploaded and blocks with deletion marks concurrently. It does that at the beginning and then every `--compact.cleanup-interval` time period. By default it is 5 minutes.
 - [#3312](https://github.com/thanos-io/thanos/pull/3312) s3: add list_objects_version config option for compatibility.
+- [#3378](https://github.com/thanos-io/thanos/pull/3378) Ruler: learned to send queries via POST. Helps when alerting/recording rules are extra long. It now uses POST unless `--query.http-method` is `GET`.
```
Comment (bwplotka) [inline CHANGELOG.md:23]: learned? ;p Can we use more technical language?

**Review by bwplotka — APPROVED** [review]
Amazing :hugs: 

Once suggestion and LGTM! :+1:

File: CHANGELOG.md (hunk 1 above)
Comment (GiedriusS) [reply CHANGELOG.md]: :smile: reworked this line to make it more technical and even cleaner. Let me know if that works :P

**Review by bwplotka — APPROVED** [review]

**Comment by bwplotka** [comment]
Thanks!

**Pull request merged**